hammer/hammer.go

```diff
@@ -97,12 +100,52 @@ func main() {
 	hammer := NewHammer(&tracker, f.Fetch, w.Write, gen, ha.seqLeafChan, ha.errChan)
 	hammer.Run(ctx)
 
+	exitCode := 0
+	if *leafWriteGoal > 0 {
+		go func() {
+			startTime := time.Now()
+			goal := tracker.LatestConsistent.Size + uint64(*leafWriteGoal)
```

Comment:
Is there a race here since `hammer.Run()` has already been called?
(Probably not _all that important_ if there is, but might lead to a "huh, why 2504?!" moment if the leaf generators were ever augmented to stop after creating `*leafWriteGoal` leaves.)

Reply:
Fixed by moving the hammer run until after this block.